docs/docsite/rst/dev_guide/developing_modules_general.rst

```diff
@@ -48,6 +48,133 @@ Creating a development environment (platform-agnostic steps)
+
+    short_description: This is my test info module
+
+    version_added: "2.4"
+
+    description:
+        - "This is my longer description explaining my test info module."
+
+    options:
+        name:
+            description:
+                - This is the message to send to the test module.
+            required: true
+            type: str
+
+    author:
+        - Your Name (@yourGitHubHandle)
+    '''
+
+    EXAMPLES = r'''
+    # Pass in a message
+    - name: Test with a message
+      my_test_info:
+        name: hello world
+    '''
+
+    RETURN = r'''
+    original_message:
+        description: The original name param that was passed in
+        type: str
+        returned: always
+    message:
+        description: The output message that the test module generates
+        type: str
+        returned: always
+    my_useful_info:
+        description: The dictionary containing information about your system
+        type: dict
+        return always
+    '''
+
+    from ansible.module_utils.basic import AnsibleModule
+
+    def run_module():
+        # define available arguments/parameters a user can pass to the module
+        module_args = dict(
+            name=dict(type='str', required=True),
+        )
+
+        # seed the result dict in the object
+        # we primarily care about changed and state
+        # changed is if this module effectively modified the target
+        # state will include any data that you want your module to pass back
+        # for consumption, for example, in a subsequent task
+        result = dict(
+            changed=False,
+            original_message='',
+            message='',
+            my_useful_info={},
+        )
+
+        # the AnsibleModule object will be our abstraction working with Ansible
+        # this includes instantiation, a couple of common attr would be the
+        # args/params passed to the execution, as well as if the module
+        # supports check mode
+        module = AnsibleModule(
+            argument_spec=module_args,
+            supports_check_mode=True
+        )
+
+        # if the user is working with this module in only check mode we do not
+        # want to make any changes to the environment, just return the current
+        # state with no modifications
+        if module.check_mode:
+            module.exit_json(**result)
+
+        # manipulate or modify the state as needed (this is going to be the
+        # part where your module will do what it needs to do)
+        result['original_message'] = module.params['name']
+        result['message'] = 'goodbye'
+        result['my_useful_info'] = {
+            'foo': 'bar',
+            'answer': 42,
+        }
+        # in the event of a successful module execution, you will want to
+        # simple AnsibleModule.exit_json(), passing the key/value results
+        module.exit_json(**result)
+
+    def main():
+        run_module()
+
+    if __name__ == '__main__':
+        main()
+
```

Comment:
How about a few words of how a facts module differs from an info module? I.e. usually less / no module options, and return values inside `ansible_facts`, and also using `ansible_facts` in `RETURNS`? Maybe it's better two have two separate examples.